src/momento/auth/momento_endpoint_resolver.py

```diff
@@ -11,27 +14,53 @@
 _CACHE_ENDPOINT_CLAIM_ID = "c"
 
 
-class _Endpoints:
-    def __init__(self, control_endpoint: str, cache_endpoint: str):
-        self.control_endpoint = control_endpoint
-        self.cache_endpoint = cache_endpoint
+@dataclass
+class _TokenAndEndpoints:
+    control_endpoint: str
+    cache_endpoint: str
+    auth_token: str
 
 
-def resolve(auth_token: str, endpoint_override: Optional[str] = None) -> _Endpoints:
-    if endpoint_override is not None:
-        return _Endpoints(
-            _MOMENTO_CONTROL_ENDPOINT_PREFIX + endpoint_override,
-            _MOMENTO_CACHE_ENDPOINT_PREFIX + endpoint_override,
+@dataclass
+class Base64DecodedToken:
+    api_key: str
+    endpoint: str
+
+
+def resolve(auth_token: Union[bytes, str]) -> _TokenAndEndpoints:
+    if not auth_token:
+        raise InvalidArgumentException("malformed auth token")
+
+    if _is_base64(auth_token):
+        decoded_b64_token = base64.b64decode(auth_token).decode("utf-8")
+        info = json.loads(decoded_b64_token)  # type: ignore[misc]
+        return _TokenAndEndpoints(
+            control_endpoint=_MOMENTO_CONTROL_ENDPOINT_PREFIX + info["endpoint"],  # type: ignore[misc]
+            cache_endpoint=_MOMENTO_CACHE_ENDPOINT_PREFIX + info["endpoint"],  # type: ignore[misc]
+            auth_token=info["api_key"],  # type: ignore[misc]
         )
-    return _get_endpoint_from_token(auth_token)
+    else:
+        if isinstance(auth_token, (bytes, bytearray)):
+            auth_token = auth_token.decode("utf-8")
+        return _get_endpoint_from_token(auth_token)
 
 
-def _get_endpoint_from_token(auth_token: str) -> _Endpoints:
+def _get_endpoint_from_token(auth_token: str) -> _TokenAndEndpoints:
     try:
         claims = jwt.decode(auth_token, options={"verify_signature": False})  # type: ignore[misc]
-        return _Endpoints(
-            claims[_CONTROL_ENDPOINT_CLAIM_ID],  # type: ignore[misc]
-            claims[_CACHE_ENDPOINT_CLAIM_ID],  # type: ignore[misc]
+        return _TokenAndEndpoints(
+            control_endpoint=claims[_CONTROL_ENDPOINT_CLAIM_ID],  # type: ignore[misc]
+            cache_endpoint=claims[_CACHE_ENDPOINT_CLAIM_ID],  # type: ignore[misc]
+            auth_token=auth_token,
         )
     except (DecodeError, KeyError) as e:
         raise InvalidArgumentException("Invalid Auth token.") from e
+
+
+def _is_base64(value: Union[bytes, str]) -> bool:
+    try:
+        if isinstance(value, str):
+            value = value.encode("utf-8")
+        return base64.b64encode(base64.b64decode(value)) == value
+    except (Exception,):
```

Comment:
what's up with the tuple here?

Reply:
This is just to suppress an IDE warning about `Exception` being overly broad. It's equivalent to `except Exception`.